pkg/providers/cloudstack/cloudstack.go

```diff
@@ -248,11 +250,7 @@ func (p *cloudstackProvider) UpdateKubeConfig(_ *[]byte, _ string) error {
 }
 
 func (p *cloudstackProvider) BootstrapClusterOpts() ([]bootstrapper.BootstrapClusterOption, error) {
-	execConfig, err := decoder.ParseCloudStackSecret()
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse environment variable exec config: %v", err)
-	}
-	return common.BootstrapClusterOpts(execConfig.ManagementUrl, p.clusterConfig)
+	return common.BootstrapClusterOpts(p.datacenterConfig.Spec.ManagementApiEndpoint, p.clusterConfig)
```

Comment:
This is a new change right? Is this a change where we are actually reading the value from our spec now instead of the secret? Does this warrant a separate PR or is it just this simple change?

Reply:
This is a simple change in my opinion. We already have validation in validator.go to ensure the values are the same